scripts/fix_nightly_feature_flags.py

```diff
@@ -0,0 +1,287 @@
+#!/bin/python3
+
+# This script checks that nightly feature flags are declared consistently so
+# we avoid nasty subtle bugs. See the individual checks below for details.
+
+import toml
+import re
+import difflib
+import sys
+
+
+class Crate:
+
+    def __init__(self, dir):
+        self.dir = dir
+        self.height = None
+        self.load_toml()
+
+    def load_toml(self):
+        self.toml = toml.load(self.dir + "/Cargo.toml")
+        self.name = self.toml["package"]["name"]
+        self.features = self.toml["features"] if 'features' in self.toml else {}
+
+        self.has_nightly_feature = "nightly" in self.features
+        self.has_nightly_protocol_feature = "nightly_protocol" in self.features
+        self.local_nightly_features = self.get_local_nightly_features()
+        self.local_nightly_protocol_features = self.get_local_nightly_protocol_features(
+        )
+        self.dependency_nightly_features = self.get_dependency_nightly_features(
+        )
+        self.dependency_nightly_protocol_features = self.get_dependency_nightly_protocol_features(
+        )
+
+    def get_local_nightly_features(self):
+        if not self.has_nightly_feature:
+            return []
+        return [
+            feature for feature in self.features["nightly"]
+            if '/' not in feature
+        ]
+
+    def get_local_nightly_protocol_features(self):
+        if not self.has_nightly_protocol_feature:
+            return []
+        return [
+            feature for feature in self.features["nightly_protocol"]
+            if '/' not in feature
+        ]
+
+    def get_dependency_nightly_features(self):
+        if not self.has_nightly_feature:
+            return []
+        return [
+            feature for feature in self.features["nightly"] if '/' in feature
+        ]
+
+    def get_dependency_nightly_protocol_features(self):
+        if not self.has_nightly_protocol_feature:
+            return []
+        return [
+            feature for feature in self.features["nightly_protocol"]
+            if '/' in feature
+        ]
+
+    def build_deps(self, crates_by_name):
+        self.deps = [
+            crates_by_name[dep]
+            for dep in self.toml['dependencies'].keys()
+            if dep in crates_by_name
+        ]
+
+    def build_transitive_deps(self):
+        visited = set()
+        result = []
+
+        def helper(crate):
```

Comment:
nit: can we use a more descriptive name? Idk, maybe `visit_recursively` or `recursion_helper` or something like that.